### DISCUSSION TITLE
The login to GitHub failed. Please try again

### DISCUSSION THREAD

**Discussion by june1888** [community/community#17617] [Copilot Conversations]
I followed those steps on https://github.com/github/copilot-docs/blob/main/docs/jetbrains/gettingstarted.md#getting-started-with-github-copilot-in-jetbrains Login to GitHub Copilot using the device auth flow and authorized GitHub Copilot IntelliJ plugin with your GitHub Account in an external browser.
but it shows The login to GitHub failed. Please try again, how can I fix this, thank you
![Screen Shot 2022-06-01 at 10 20 30](https://user-images.githubusercontent.com/97771780/171440193-d55d12a5-1736-458d-b64b-91bb71732df8.png)

**Comment by jansorg** [comment]
Are you using a proxy server?

**Reply by june1888** [reply]
No, and I download version 14.17 from [https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable](https://github.com/github/feedback/discussions/url), but still does not work

**Reply by nullsss** [reply]
The same problem

**Reply by jansorg** [reply]
@june1888 Which version of the plugin do you have installed (`Preferences > Plugins > Installed Plugins`)? Could you please try to update to the latest version (1.1.24.1610)?

**Reply by JunQu** [reply]
same problem. i just updated the plugin on webstorm eap, it's version show `1.1.24.1610`.my vscode copilot plugin is working fine until now.
it seems that i need to login again after updating copilot plugin to encounter this problem.
<img width="384" alt="image" src="https://user-images.githubusercontent.com/39846309/171586018-af3d9e5b-7603-4457-b89b-449778e38ff1.png">

i use chrome (Version 102.0.5005.61 (Official Build) (x86_64)) and webstorm (2022.2 EAP build#ws-222.2680.4)

**Reply by djcraft** [reply]
Hello,
I have the same problem since the last update on `IDEA2022.1.1`.
I have one difference though, I have a corporate proxy that I don't manage that only allows HTTP/HTTPS access.

I am of course on version `1.1.24.1610`.

**Reply by d15801543974** [reply]
![image](https://user-images.githubusercontent.com/23231997/171588477-5f74bde9-5be8-403a-b25f-fd66b11f0c12.png)

**Reply by d15801543974** [reply]
Failed to log in after updating to the latest version. It is used normally in the previous version。
GitHub Copilot 1.1.24.1610        IntelliJ IDEA 2021.2.2。
Use network proxy   Same question

**Reply by djcraft** [reply]
I retrieved the idea logs and it appears that there is an ETIMEOUT on the IP with an error code -32603 at the time of the connection request

**Reply by d15801543974** [reply]
我的现在可以了，就是一直尝试。说不定哪次就可以了。随缘吧

**Reply by shrinidhinhegde** [reply]
I have been having the same issue for 2 weeks now (both on webstorm and pycharm). I have sort of given up on copilot and coding on my own like in the old days lol. 
I tried:
- uninstalling and reinstalling the plugin from the preferences>plugins 
- uninstalled webstorm and reinstalled again (yes i was very desperate :/ )


i have not been using proxy or vpn for that matter.

pls lmk if i am doing anything wrong.

**Reply by yagamiy1023** [reply]
> ![图片](https://user-images.githubusercontent.com/23231997/171588477-5f74bde9-5be8-403a-b25f-fd66b11f0c12.png)

我也一直卡在这里，你解决了吗

**Reply by ShawnZhong** [reply]
I can log in with `1.1.20.1417`, but the completion does not work with the following error message. 

<img width="372" alt="image" src="https://user-images.githubusercontent.com/6421097/172797485-3419f7c0-bddf-4151-9f53-bcb810cfa604.png">

**Reply by Moses-Code-Dev** [reply]
This worked for me
https://github.com/github-community/community/discussions/16965#discussioncomment-2916107

**Reply by Xuan-1998** [reply]
> I can log in with `1.1.20.1417`, but the completion does not work with the following error message.
> 
> <img alt="image" width="372" src="https://user-images.githubusercontent.com/6421097/172797485-3419f7c0-bddf-4151-9f53-bcb810cfa604.png">

I have the same issue @ShawnZhong did you solve it?

**Comment by shrinidhinhegde** [comment]
So I got it working by downloading THIS version of the copilot plugin from jetbrains (https://plugins.jetbrains.com/plugin/17718-github-copilot/versions/stable/172765)


and installed the plugin from disk and tried to log in... and *drum rolls* it was successful. 

from now on I would not plan to auto-update the plugin.

**Reply by bmuskalla** [reply]
> from now on I would not plan to auto-update the plugin.

Let's work on getting your setup to work on newer version as we'll have to phase out older plugin versions over time due to API breakages. Can you check if you see any copilot related things in the IntelliJ logs? Without any further hint, it's really hard to for us to reproduce. And for sanity check, you can verify which engine copilot is using (in the plugin preferences?) 
And maybe exactly what kind of problem you're seeing? There is a variety of (maybe related) problems discussed in this thread.

**Reply by shrinidhinhegde** [reply]
I can see from the IntelliJ logs, i am having a timeout error [ETIMEOUT on the IP with an error code -32603] while i try to login on any version after 1.20.xxxx.

and now 1.20 seems to be phased out.

to reproduce this problem, just install the plugin on webstorm/pycharm and you should basically face the same issue i am facing.

**Reply by bmuskalla** [reply]
@shrinidhinhegde Indeed, a few things in our network stack have changed from 1.20.xxx onwards. The new versions currently don't support a proxy setup. Would you please point out what kind of proxy setup your environment need to access `github.com`? 
Can you try to see if you can access `github.com` without a proxy with something like
```
curl --noproxy '*' github.com
```
And can you please share more details what exact steps work and don't work (do you get to the point that you have a device code? did you enter it on github.com?)

**Reply by shrinidhinhegde** [reply]
```curl --noproxy '*' github.com```


does not return anything on my terminal.. I do not have any proxy setup.. in my ide, this is how the proxy settings look.
<img width="618" alt="Screenshot 2022-06-07 at 18 19 14" src="https://user-images.githubusercontent.com/46134463/172383085-817c91b5-7e71-49a4-adae-08e272faf566.png">

and i am able to install the plugin and when i click on login to github.. i get the device code..
<img width="422" alt="Screenshot 2022-06-07 at 18 16 21" src="https://user-images.githubusercontent.com/46134463/172382252-fb1c1143-829c-4fa8-aed7-85ebc4c91ec8.png">


and then when i click 'Copy and Open' and fill in the device code on my browser... and authorize.... i get  the following
<img width="660" alt="Screenshot 2022-06-07 at 18 17 48" src="https://user-images.githubusercontent.com/46134463/172382569-848c1bde-5a75-47da-b998-a38f5a7b050b.png">

but back in the ide... i get this

<img width="372" alt="Screenshot 2022-06-07 at 18 18 24" src="https://user-images.githubusercontent.com/46134463/172382759-2f3265da-e1af-4402-ba6e-9c66276d7bed.png">


currently the github copilot version is 1.1.24.1610

**Reply by bmuskalla** [reply]
Thanks for the detailed response. I know this is a lot to ask but would you be so kind to try if Copilot works for you within Visual Studio Code? I'm trying to narrow down which part of the system is failing for you and all signal I have right now is inconclusive. I'm still digging into the data to see if there is something to get you set up again

**Reply by djcraft** [reply]
Looking at the various threads, the error seems to come from the plugin's inability to go through the proxy configured in the IDE.

**Reply by shrinidhinhegde** [reply]
@bmuskalla it works with vs code. copilot version is 1.27.6094

**Reply by shrinidhinhegde** [reply]
@djcraft but i do not have any proxy configured on the ide like i mentioned in this thread.

**Reply by djcraft** [reply]
@shrinidhinhegde I'm sorry I didn't have the first message in the message (problem connecting to github because of my proxy precisely)

**Reply by gal-leibovich** [reply]
> @shrinidhinhegde Indeed, a few things in our network stack have changed from 1.20.xxx onwards. The new versions currently don't support a proxy setup. Would you please point out what kind of proxy setup your environment need to access `github.com`? Can you try to see if you can access `github.com` without a proxy with something like
> 
> ```
> curl --noproxy '*' github.com
> ```
> 
> And can you please share more details what exact steps work and don't work (do you get to the point that you have a device code? did you enter it on github.com?)

Removing the proxy support messed things up for me as well. I'm on 1.1.24.1610. 
`curl --noproxy '*' github.com` doesn't work either.

Do you have plans to restore proxy support? as previous versions of copilot are now said to be unsupported.

**Reply by olbod** [reply]
> @shrinidhinhegde Indeed, a few things in our network stack have changed from 1.20.xxx onwards. The new versions currently don't support a proxy setup. Would you please point out what kind of proxy setup your environment need to access `github.com`? Can you try to see if you can access `github.com` without a proxy with something like
> 
> ```
> curl --noproxy '*' github.com
> ```
> 
> And can you please share more details what exact steps work and don't work (do you get to the point that you have a device code? did you enter it on github.com?)

I have exactly the same behavior as @shrinidhinhegde described. Checked --noproxy also for `api.github.com` and `github.global.ssl.fastly.net`. No errors.

**Reply by shrinidhinhegde** [reply]
@olbod How do you set no proxy for particular urls? Can you provide a detailed answer please?

**Reply by olbod** [reply]
> @olbod How do you set no proxy for particular urls? Can you provide a detailed answer please?

I have used the same command `curl --noproxy '*' github.com` in the terminal. I just replaced the URL with `api.github.com` etc.. When you execute the command and see no errors it means that you can reach the URL without proxy.

**Reply by shrinidhinhegde** [reply]
@olbod i did try to login after set no proxy for the urls u mentioned.. no success...

**Reply by olbod** [reply]
@shrinidhinhegde Oh, got you. I posted not a solution for the bug. I still have `The login to GitHub failed ...' issue. The command that I have executed just means that there are no proxy errors and I can reach all required for Copilot APIs, but still can't login.

**Reply by shrinidhinhegde** [reply]
oh! my bad

**Reply by Xiao-Chenguang** [reply]
same problem. copilot works on vscode but not pycharm with my Mac. There is also no proxy in my device and GitHub.com can be accessed directly.

**Reply by shrinidhinhegde** [reply]
@bmuskalla any good news?

**Reply by Moses-Code-Dev** [reply]
@shrinidhinhegde
https://github.com/github-community/community/discussions/18132#discussioncomment-2918095

**Reply by olbod** [reply]
> @shrinidhinhegde [#18132 (comment)](https://github.com/orgs/github-community/discussions/18132#discussioncomment-2918095)

@djoxy-code Thanks! The solution works for me. The only thing for mac path to replace will be ~/Library/Application Support/JetBrains/PyCharm2022.1/plugins/github-copilot-intellij/lib (for my pycharm 2022.1).

**Comment by Akalanka1337** [comment]
Hello,
Use "https://github.com/settings/tokens" to login.It should work!

**Reply by baruchaniview** [reply]
@Akalanka1337 how can I provide the token to Copilot? Could you please elaborate?

**Comment by baruchaniview** [comment]
Having the same issue on Android Studio with Copilot 1.1.28.1744 new installation. The browser says success but the Android Studio says failure.
![Screen Shot 2022-07-04 at 12 27 54](https://user-images.githubusercontent.com/94789971/177128462-bb91172e-03f3-4c40-82df-e562bf7ebb63.png)

**Reply by Moses-Code-Dev** [reply]
@baruchaniview 
try this 
https://github.com/orgs/github-community/discussions/18132#discussioncomment-2918095

and make sure to "set" the Suggestions matching public code located in https://github.com/settings/copilot.

**Reply by baruchaniview** [reply]
Thanks @djoxy-code it worked.
Should note for others that I'm using a mac and the plugins are located at 
`~/Library/Application Support/Google/AndroidStudio<your version>/plugins/github-copilot-intellij`
and had to use [BetterZip Quick Look](https://macitbetter.com/BetterZip-Quick-Look-Generator/) to view the content of the jar.

**Comment by shrinidhinhegde** [comment]
So looks like the issue is still not fixed.. everytime there is an update.. i find myself doing this.. 

https://github.com/orgs/github-community/discussions/18132#discussioncomment-2918095

can we please have a fix as github co-pilot is a paid product now?

**Comment by baruchaniview** [comment]
So it seems replying to my own thread isn't working so starting a new one.
Updated to a new version (1.1.29.1869) and got locked out again. 
The solution described here that helped me login in the previous version doesn't work anymore.
(This one: https://github.com/orgs/github-community/discussions/18132#discussioncomment-2918095).
Can anyone please assist?

Using Android Studio 2021.2.2 on a Mac.

![copilot_fail](https://user-images.githubusercontent.com/94789971/184132209-b1b98d3f-7808-46a6-bf6e-82d544829dcd.png)

**Reply by shrinidhinhegde** [reply]
Hey I got an update today (1.1.29.1869) and everything works fine for me. I had issues until the previous update.  i am using the jetbrains toolbox version of webstorm and pycharm (if that makes any difference at all)

**Reply by baruchaniview** [reply]
Got another update today - 1.1.30.1887. Still the same issue.
Just unbelievable. A paid product from a big company with such bugs for so long and zero support.
Very disappointing!

**Reply by baruchaniview** [reply]
finally found a solution by exploring the IDE logs. 

Need to run the following command in the terminal: ```sudo chmod 777 ~/.config```
(777 might be an overkill)

Here is the error:

> 2022-08-18 14:24:16,273 [  31572]   WARN - tions.ex.ConfigurableCardPanel - auto-dispose 'Plugins' id=preferences.pluginManager 
> 2022-08-18 14:24:38,728 [  54027]   WARN - .lang.agent.AgentGitHubService - error polling for GitHub token 
> com.github.copilot.lang.agent.rpc.JsonRpcErrorException: Error processing command signInConfirm, request id: 5, error code: 1001
> 	at com.github.copilot.lang.agent.vscodeRpc.DefaultJsonRpcMessageHandler.handleErrorResponse(DefaultJsonRpcMessageHandler.java:195)
> 	at com.github.copilot.lang.agent.vscodeRpc.DefaultJsonRpcMessageHandler.handleJsonMessage(DefaultJsonRpcMessageHandler.java:74)
> 	at com.github.copilot.lang.agent.vscodeRpc.VSCodeJsonRpcParser.processPendingContent(VSCodeJsonRpcParser.java:91)
> 	at com.github.copilot.lang.agent.vscodeRpc.VSCodeJsonRpcParser.appendLocked(VSCodeJsonRpcParser.java:71)
> 	at com.github.copilot.lang.agent.vscodeRpc.VSCodeJsonRpcParser.append(VSCodeJsonRpcParser.java:39)
> 	at com.github.copilot.lang.agent.vscodeRpc.AgentProcessHandler$1.onTextAvailable(AgentProcessHandler.java:35)
> 	at jdk.internal.reflect.GeneratedMethodAccessor382.invoke(Unknown Source)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
> 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
> 	at com.intellij.execution.process.ProcessHandler$2.invoke(ProcessHandler.java:247)
> 	at com.sun.proxy.$Proxy35.onTextAvailable(Unknown Source)
> 	at com.intellij.execution.process.ProcessHandler.notifyTextAvailable(ProcessHandler.java:221)
> 	at com.intellij.execution.process.BaseOSProcessHandler$SimpleOutputReader.onTextAvailable(BaseOSProcessHandler.java:156)
> 	at com.intellij.util.io.BaseOutputReader.processInput(BaseOutputReader.java:198)
> 	at com.intellij.util.io.BaseOutputReader.readAvailableBlocking(BaseOutputReader.java:136)
> 	at com.intellij.util.io.BaseDataReader.readAvailable(BaseDataReader.java:77)
> 	at com.intellij.util.io.BaseDataReader.doRun(BaseDataReader.java:166)
> 	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:213)
> 	at com.intellij.util.io.BaseDataReader.lambda$start$0(BaseDataReader.java:51)
> 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
> 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
> 	at java.base/java.security.AccessController.doPrivileged(Native Method)
> 	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
> 	at java.base/java.lang.Thread.run(Thread.java:829)
> Caused by: com.github.copilot.lang.agent.rpc.JsonRpcErrorException: Error: EACCES: permission denied, mkdir '/Users/<username>/.config/github-copilot', request id: 5, error code: 1001
> 	at com.github.copilot.lang.agent.rpc.JsonRPC.parseResponse(JsonRPC.java:61)
> 	at com.github.copilot.lang.agent.vscodeRpc.DefaultJsonRpcMessageHandler.handleJsonMessage(DefaultJsonRpcMessageHandler.java:72)
> 	... 26 more

**Reply by bmuskalla** [reply]
Thanks for digging into this. I've forward this internally so problems like this produce better error messages in the future. Thanks again for your persistence and sharing the solution with others.

**Comment by shirlston** [comment]
> finally found a solution by exploring the IDE logs.
> 
> Need to run the following command in the terminal: `sudo chmod 777 ~/.config` (777 might be an overkill)

💯 This worked! 777 probably is overkill, but this is the only solution that works for me in RubyMine 2022.2.1.

So after being a beta tester from the very beginning, but never having been able to use it, I now have 2 days to decide if I like copilot or not.

**Comment by Xuan-1998** [comment]
<img width="233" alt="image" src="https://user-images.githubusercontent.com/58761221/188484009-2680ac6a-0cdd-4682-b42b-8f8d3d26259a.png">
how to solve this problem

**Comment by davidesciotti** [comment]
I don't know if it's relevant, but after trying many different things I solved creating manually the folder `github-copilot` in `/Users/<username>/.config/`. I am working with PyCharm on an M1 mac. 
I hope it's helpful!

**Comment by joeflack4** [comment]
I [exported my PyCharm settings](https://www.jetbrains.com/help/mps/exporting-and-importing-settings.html), [did an uninstall](https://www.jetbrains.com/help/pycharm/uninstall.html#163013c4), reinstalled PyCharm, reinstalled GitHub copilot plugin, restarted IDE, and now it finally allowed me to sign in and authorize.

However this problem remains ;(

I just installed the plugin in IntelliJ and it works fine, but for PyCharm it still does not work.

---

**Edit: Fixed!**

Removing PyCharm folders from these was not sufficient:
```
~/Library/Application Support/JetBrains/<product><version>
~/Library/Caches/JetBrains/<product><version>
```

I also had to remove PyCharm folders from ([thread](https://intellij-support.jetbrains.com/hc/en-us/community/posts/206593875-Complete-uninstall-on-Mac)):
```
~/Library/Preferences/<product><version>
```

**Comment by igorwojda** [comment]
### Solution (IntelliJ IDEA / Android Studio)
Based on the above @joeflack4 answer I was able to find the solution (thx @joeflack4). It also requires file removal, but it limits the removal scope only to copilot config files (exit IDE before removing files):

IntelliJ IDEA:
`~/Library/Application Support/JetBrains/IDEA_VERSION/options/github-copilot.xml`
`~/Library/Application Support/JetBrains/IDEA_VERSION/options/github-copilot.local.xml`

or

AndroidStudio:
`~/Library/Application Support/Google/ANDROID_STUDIO_VERSION/options/github-copilot.xml`
`~/Library/Application Support/Google/ANDROID_STUDIO_VERSION/options/github-copilot.local.xml`

Now Open ID, open GithubPilot view and hit `refresh` (this may require GitHub login in the browser)

### Note For Copilot Devs
I hope this will help you with fixing the plugin flow. 
Version: `1.2.3.2385` 

Here are old and new files (the new ones are for IDEA only as I could not find them for AS). It looks like this may have something to do with `githubTokenMigration` attribute:

Android Studio removed `github-copilot.local`.xml:
```xml
<application>
  <component name="github-copilot-local">
    <option name="lastUpdateCheck" value="2022-12-13T10:57:11.28032+01:00[Europe/Warsaw]" />
    <option name="githubTokenMigration" value="true" />
  </component>
</application>
```

Android Studio removed `github-copilot.xml`:
```xml
<application>
  <component name="github-copilot">
    <option name="signinNotificationShown" value="true" />
  </component>
</application>
```

IntelliJ IDEA new `github-copilot.local`.xml:
```xml
<application>
  <component name="github-copilot">
    <option name="signinNotificationShown" value="true" />
  </component>
</application>
```

IntelliJ IDEA removed `github-copilot.local.xml`:
```xml
<application>
  <component name="github-copilot-local">
    <option name="lastUpdateCheck" value="2023-02-21T12:08:20.258333+01:00[Europe/Warsaw]" />
    <option name="githubTokenMigration" value="true" />
  </component>
</application>
```

IntelliJ IDEA removed `github-copilot.xml`:
```xml
<application>
  <component name="github-copilot">
    <option name="signinNotificationShown" value="true" />
    <option name="showIdeCompletions" value="true" />
  </component>
</application>
```
